core/src/main/java/com/transferwise/common/gracefulshutdown/strategies/TaskSchedulersGracefulShutdownStrategy.java

```diff
@@ -1,86 +1,53 @@
 package com.transferwise.common.gracefulshutdown.strategies;
 
 import com.transferwise.common.gracefulshutdown.GracefulShutdownStrategy;
+import com.transferwise.common.gracefulshutdown.utils.ShutdownUtils;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.ApplicationContext;
+import org.springframework.scheduling.TaskScheduler;
+
 import java.util.ArrayList;
 import java.util.HashSet;
 import java.util.List;
-import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executor;
 import java.util.concurrent.Executors;
-import java.util.concurrent.ScheduledThreadPoolExecutor;
 import java.util.concurrent.atomic.AtomicInteger;
-import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.ApplicationContext;
-import org.springframework.scheduling.TaskScheduler;
-import org.springframework.scheduling.concurrent.ConcurrentTaskScheduler;
-import org.springframework.scheduling.concurrent.ThreadPoolTaskScheduler;
 
 @Slf4j
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder(toBuilder = true)
 public class TaskSchedulersGracefulShutdownStrategy implements GracefulShutdownStrategy {
 
   @Autowired
   private ApplicationContext applicationContext;
 
-  private List<TaskScheduler> taskSchedulers = new ArrayList<>();
+  private final List<TaskScheduler> taskSchedulers = new ArrayList<>();
 
-  private AtomicInteger inProgressShutdowns = new AtomicInteger();
+  private final AtomicInteger inProgressShutdowns = new AtomicInteger();
 
   @Override
   public void prepareForShutdown() {
     var executors = Executors.newFixedThreadPool(10);
 
     var taskSchedulerBeans = applicationContext.getBeansOfType(TaskScheduler.class).values();
-    var allTaskSchedulers = new HashSet(taskSchedulerBeans);
+    var allTaskSchedulers = new HashSet<>(taskSchedulerBeans);
     allTaskSchedulers.addAll(taskSchedulers);
 
     for (var taskSchedulerProto : allTaskSchedulers) {
       inProgressShutdowns.incrementAndGet();
       executors.submit(() -> {
         var taskScheduler = taskSchedulerProto;
         try {
-          if (taskScheduler instanceof ThreadPoolTaskScheduler) {
```

Comment:
Behaviour is same as before. Extracted reusable logic to ShutdownUtils. Only log messages was slightly changed with `task scheduler` text replaced with `executor`.